src/main/java/seedu/address/model/localcourse/LocalName.java

```diff
@@ -0,0 +1,62 @@
+package seedu.address.model.localcourse;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.address.commons.util.AppUtil.checkArgument;
+
+/**
+ * Represents a Local Course's local name in SEPlendid.
+ * Guarantees: immutable; is valid as declared in {@link #isValidLocalName(String)}
+ */
+public class LocalName {
+
+    // Used in AppUtil#checkArgument and ParserUtil, JsonAdaptedXXX exceptions
+    public static final String MESSAGE_CONSTRAINTS = "LocalName can take any values, and it should not be blank";
+
+    /*
+     * This matches a string that starts with a non-whitespace character.
+     */
+    public static final String VALIDATION_REGEX = "[^\\s].*";
```

Comment:
Do you want it to contain [ and ]?

Reply:
Nope, `[^]` refers to negation in regex.